### ISSUE TITLE
[Bug] String values in defaults list are not overridable with lists

### ISSUE THREAD

**Issue by apsdehal** [facebookresearch/hydra#1383]
# 🐛 Bug
## Description

String values in defaults list are not overridable with lists.

## Checklist
- [x] I checked on the latest version of Hydra
- [x] I created a minimal repro (See [this](https://stackoverflow.com/help/minimal-reproducible-example) for tips).

## To reproduce

** Minimal Code/Config snippet to reproduce **

Folder structure to reproduce:

```
├── conf
│   ├── conf.yaml
│   └── dataset
│       ├── textcaps.yaml
│       └── textvqa.yaml
├── main.py
```

Contents:

1. main.py

```py
import hydra
from omegaconf import DictConfig

@hydra.main(config_path="conf", config_name="conf")
def my_app(cfg : DictConfig) -> None:
    print(cfg)
    
if __name__ == "__main__":
    my_app()
```

2. conf.yaml

```yaml
defaults:
- dataset: textvqa
```

3. conf/dataset/textvqa.yaml
```yaml
textvqa:
  b: c
```

4. conf/dataset/textcaps.yaml

```yaml
textcaps:
  a: b
```

### Command

```bash
python main.py dataset=[textvqa,textcaps]
```

**Stack trace/error message**
```
Traceback (most recent call last):
  File "/my_hydra_clone/hydra/hydra/_internal/utils.py", line 212, in run_and_report
    return func()
  File "/my_hydra_clone/hydra/hydra/_internal/utils.py", line 368, in <lambda>
    overrides=args.overrides,
  File "/my_hydra_clone/hydra/hydra/_internal/hydra.py", line 104, in run
    run_mode=RunMode.RUN,
  File "/my_hydra_clone/hydra/hydra/_internal/hydra.py", line 544, in compose_config
    from_shell=from_shell,
  File "/my_hydra_clone/hydra/hydra/_internal/config_loader_impl.py", line 156, in load_configuration
    from_shell=from_shell,
  File "/my_hydra_clone/hydra/hydra/_internal/config_loader_impl.py", line 187, in _load_configuration_impl
    skip_missing=run_mode == RunMode.MULTIRUN,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 720, in create_defaults_list
    skip_missing=skip_missing,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 691, in _create_defaults_list
    skip_missing=skip_missing,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 343, in _create_defaults_tree
    overrides=overrides,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 420, in _create_defaults_tree_impl
    return _expand_virtual_root(repo, root, overrides, skip_missing)
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 268, in _expand_virtual_root
    overrides=overrides,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 539, in _create_defaults_tree_impl
    add_child(children, new_root)
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 491, in add_child
    overrides=overrides,
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 439, in _create_defaults_tree_impl
    update_package_header(repo=repo, node=parent)
  File "/my_hydra_clone/hydra/hydra/_internal/defaults_list.py", line 238, in update_package_header
    loaded = repo.load_config(config_path=node.get_config_path())
  File "/my_hydra_clone/hydra/hydra/core/default_element.py", line 470, in get_config_path
    return f"{group_path}/{self.get_name()}"
  File "/my_hydra_clone/hydra/hydra/core/default_element.py", line 479, in get_name
    assert self.value is None or isinstance(self.value, str)
AssertionError
```

## Expected Behavior
Should work and result in config:
```
{'dataset': {'textvqa': {'b': 'c'}, 'textcaps': {'a': 'b'}}}
```

## System information
- **Hydra Version** :  Master
- **Python version** : 3.7
- **Virtual environment type and version** : conda 4.5.8
- **Operating system** : Ubuntu 18.04

## Additional context

Note that this works in reverse way, if the defaults list value is a list and you override it with a string.

**Comment by omry** [comment]
Thanks for reporting!

**Comment by omry** [comment]
@apsdehal, got a fix ready (#1401).
seems to work on your example, let me know if you run into anything.
(Will merge in later today).
